### ISSUE TITLE
Help and configuring YOLO correctly ?

### ISSUE THREAD

**Issue by newinai** [experiencor/keras-yolo3#182]
Hi,

I have some questions about YOLO configuration for my AI problem.

**To resume :**
- 2 classes to detect in my dataset, A with 4500 features and B with 1900 features.
- A shapes are generaly big and B shapes smaller (depends of the picture).
- Original pictures are 6000x4000px.

**Configuration:**
- 2 GPU.
- batch_size=16.
- train_times=3
- nb_epochs=100
- min_input_size=416 ; max_input_size=416

**After multiple training trials :**
- Yolo loss ~28 ; Yolo layer1 loss ~ < 1 ; Yolo layer2 loss ~ <1 ; Yolo layer3 loss: ~13

**Evaluation :**
- mAP is really bad ~ < 1
- Only A is detected

**Some questions :**
- Do i need to calculate my own dataset anchors ? If yes, is it compatible with my original size 6000px*4000px ?
- Is there a relation between anchors and min_input_size and max_input_size ?
- Do i need to increase min_input_size and max_input_size ?
- Why my loss not decreasing more ?
- Is it possible to print val_loss ? And is it a good thing to print it ? If no, when to stop training with only loss ?
- What do you suggest more ?

Thank you in advance !

**Comment by kabrau** [comment]
Do I need to calculate my own dataset anchors? If yes, is it compatible with my original size 6000px*4000px ?
- I think you have a problem because the YOLO reshape your images to go through the network
- "min_input_size":       288,      
- "max_input_size":       448,

Is there a relation between anchors and min_input_size and max_input_size ?
- If you use gen_anchors.py to create anchors you do not have a problem, but, in my Fork, I change gen_acnhors.py to verify validations dataset (https://github.com/kabrau/keras-yolo3).

Do I need to increase min_input_size and max_input_size ?
- Yes

Why my loss not decreasing more ? 
- Depends, there are many aspects, bad annotations, meaningless dataset, I do not understand what is your features (2 classes to detect in my dataset, A with 4500 features and B with 1900 features),  for example, the classes are CAT or DOG, but you do not need features, the YOLO will create the features.

Is it possible to print val_loss ? And is it a good thing to print it ? If no, when to stop training with only loss ?
- Yes, I prefer to use validation dataset too, you need to change the fit_generator and add validations parameters. If you will have problems, send me a message.

What do you suggest more ? 
- Show me one sample, maybe a can help you.

**Comment by sabrinatuli** [comment]
How can I compute min_input_size and max_input_size for my dataset? I have generated my own anchors, but don't know what to do with min and max_input_size.
